### PR TITLE
feat: composable meta commands

### LINKED ISSUE #218: More expressive meta commands
We want to be able to say things like`!(def a !(chain ...))`

### PR DESCRIPTION
**Pull request by arthurpaulino**
Makes it possible to assemble meta commands in Lurk expressions. Each meta command is resolved to the value it returns before evaluating the whole expression with the VM.

Closes #218

### CODE REVIEW

**Review by wwared — COMMENTED** [review]
Looks great! non-blocking nits below

File: src/lurk/tests/eval.rs [hunk 1]
```diff
@@ -391,7 +387,7 @@ test!(
     test_letrec_error,
     "(letrec ((odd? (lambda (n) (if (= n 0) nil (even? (- n 1)))))
               (x a)
-              (even? (lambda (n) (if (= n 0) true (odd? (- n 1))))))
+              (even? (lambda (n) (if (= n 0) t (odd? (- n 1))))))
```
[the remark this answers is not in the document]
Comment (arthurpaulino) [reply src/lurk/tests/eval.rs]: #346

File: src/lurk/parser/syntax.rs [hunk 2]
```diff
@@ -418,62 +397,130 @@ pub fn parse_list<F: Field>(
...
             parse_comm(),
             parse_big_num(),
-            context(
-                "symbol",
-                parse_symbol(state.clone(), create_unknown_packages),
-            ),
+            parse_symbol_syntax(state.clone(), create_unknown_packages),
             parse_string(),
-            context("quote", parse_quote(state.clone(), create_unknown_packages)),
+            parse_char_or_quote(state.clone(), create_unknown_packages),
```
[the remark this answers is not in the document]
Comment (arthurpaulino) [reply src/lurk/parser/syntax.rs:464]: The implementation already does it. I just gave it a better name.

[the remark this answers is not in the document]
Comment (wwared) [reply src/lurk/parser/syntax.rs:464]: D'oh, you're right, good point!

**Review by wwared — APPROVED** [review]
This is very exciting, thank you!

**Pull request merged**